### PR TITLE
Feature/validator package

### PR DESCRIPTION
**Pull request by mehmet-erim**

### CODE REVIEW

File: packages/validators/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+module.exports = function (config) {
```
[the remark this answers is not in the document]
Comment (armanozak) [reply packages/validators/karma.conf.js]: Could you `require` a base configuration and extend it if necessary?

File: angular.json [hunk 2]
```diff
@@ -160,12 +179,52 @@
...
+              "**/node_modules/**"
+            ]
           }
         }
       }
     }
   },
   "defaultProject": "ngx-validate"
-}
+}
```
[the remark this answers is not in the document]
Comment (armanozak) [reply angular.json]: I am seeing a new line in my editor, but could you please double-check that?

File: tsconfig.json [hunk 3]
```diff
@@ -11,11 +11,26 @@
...
+      "validators": [
+        "dist/validators"
+      ],
+      "validators/*": [
+        "dist/validators/*"
+      ]
     }
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (armanozak) [reply tsconfig.json]: Another newline issue here.

File: packages/validators/ng-package.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
+  "dest": "../../dist/validators",
+  "lib": {
+    "entryFile": "src/public_api.ts"
+  }
+}
```
[the remark this answers is not in the document]
Comment (mehmet-erim) [reply packages/validators/ng-package.json]: bullshit

File: src/app/app.component.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
 import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
-import { comparePasswords, validatePassword, Validation } from '../../packages/core/src/public_api';
+import { validatePassword, comparePasswords } from 'packages/validators/src/public_api';
```
Comment (armanozak) [inline src/app/app.component.ts]: I am pretty sure this should be as follows:

```
import { validatePassword, comparePasswords } from '../../packages/validators/src/public_api';
```

File: packages/validators/src/lib/validators/password-validators.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import { AbstractControl, FormGroup, ValidatorFn } from '@angular/forms';
-import { addCommas, normalizeDiacritics } from '../utils';
+import { normalizeDiacritics, addCommas } from '@ngx-validate/core';
```
[the remark this answers is not in the document]
Comment (armanozak) [reply packages/validators/src/lib/validators/password-validators.ts]: `normalizeDiacritics` and `addCommas` are in the validators module, aren't they?

File: packages/validators/src/utils/string-utils.ts [hunk 7]
```diff
@@ -0,0 +1,408 @@
+import { Validation, mapReplace } from '@ngx-validate/core';
```
Comment (armanozak) [inline packages/validators/src/utils/string-utils.ts]: I think we should add an tsconfig alias similar to the following and import anything from core to validators referring to this alias:

```
"paths": {
  "core": [
    "dist/core",
    "libs/core/src/public_api.ts"
  ]
}
```

File: packages/validators/package.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@ngx-validate/validators",
+  "version": "0.0.1",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/ng-turkey/ngx-validate.git"
+  },
+  "dependencies": {
+    "@ngx-validate/core": "0.0.1"
```
Comment (armanozak) [inline packages/validators/package.json]: Please check my comments. I think we may not need this dependency after all.

File: packages/validators/ng-package.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
+  "dest": "../../dist/validators",
+  "lib": {
+    "entryFile": "src/public_api.ts"
+  },
+  "whitelistedNonPeerDependencies": [
+    "@ngx-validate/core"
```
Comment (armanozak) [inline packages/validators/ng-package.json]: If we get rid of the dependency, this as well will be unnecessary.

File: packages/validators/src/utils/string-utils.ts (hunk 7 above)
Comment (armanozak) [inline packages/validators/src/utils/string-utils.ts]: I think we should add a tsconfig alias similar to the following and import anything from core to validators referring to this alias:

```
"paths": {
  "core": [
    "dist/core",
    "libs/core/src/public_api.ts"
  ],
  "core/*": [
    "dist/core/*",
    "libs/core/src/lib/*"
  ]
}
```

File: src/app/app.component.ts (hunk 5 above)
Comment (armanozak) [inline src/app/app.component.ts]: I am pretty sure this should be as follows:

```
import { validatePassword, comparePasswords } from 'validators';
```

File: src/app/app.component.ts [hunk 10]
```diff
@@ -1,6 +1,7 @@
...
+import { Validation } from '../../packages/core/src/public_api';
```
Comment (armanozak) [inline src/app/app.component.ts]: And this as follows:

```
import { Validation } from 'core';
```

File: packages/validators/src/lib/validators/password-validators.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (armanozak) [reply packages/validators/src/lib/validators/password-validators.ts]: `normalizeDiacritics` and `addCommas` are in the validators module now, aren't they?

**Pull request closed without merging**